pkg/provider/kind/kind.go

```diff
@@ -0,0 +1,209 @@
+	c.DeploymentVars = provider.MergeDeploymentVars(
+		c.DeploymentResource.DefaultDeploymentVars,
+		customDeploymentVars,
+		c.DeploymentResource.FlagDeploymentVars,
+	)
+	return nil
+}
+
+// KINDDeploymentsParse parses the environment/kind deployment files and saves the result as bytes grouped by the filename.
+// Any variables passed to the cli will be replaced in the resources files following the golang text template format.
+func (c *KIND) KINDDeploymentsParse(*kingpin.ParseContext) error {
+	if err := c.checkDeploymentVarsAndFiles(); err != nil {
+		return err
+	}
+
+	deploymentResource, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		return err
+	}
+	c.kindResources = deploymentResource
+	return nil
+}
+
+func (c *KIND) K8SDeploymentsParse(*kingpin.ParseContext) error {
+	if err := c.checkDeploymentVarsAndFiles(); err != nil {
+		return err
+	}
+
+	deploymentResource, err := provider.DeploymentsParse(c.DeploymentFiles, c.DeploymentVars)
+	if err != nil {
+		log.Fatalf("Couldn't parse deployment files: %v", err)
+	}
+	for _, deployment := range deploymentResource {
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		k8sObjects := make([]runtime.Object, 0)
+
+		for _, text := range strings.Split(string(deployment.Content), provider.Separator) {
+			text = strings.TrimSpace(text)
+			if len(text) == 0 {
+				continue
+			}
+
+			resource, _, err := decode([]byte(text), nil, nil)
+			if err != nil {
+				return errors.Wrapf(err, "decoding the resource file:%v, section:%v...", deployment.FileName, text[:100])
+			}
+			if resource == nil {
+				continue
+			}
+			k8sObjects = append(k8sObjects, resource)
+		}
+		if len(k8sObjects) > 0 {
+			c.k8sResources = append(c.k8sResources, k8sProvider.Resource{FileName: deployment.FileName, Objects: k8sObjects})
+		}
+	}
+	return nil
+}
+
+// checkDeploymentVarsAndFiles checks whether the requied deployment vars are passed.
+func (c *KIND) checkDeploymentVarsAndFiles() error {
+	reqDepVars := []string{"PR_NUMBER", "CLUSTER_NAME"}
+	for _, k := range reqDepVars {
+		if v, ok := c.DeploymentVars[k]; !ok || v == "" {
+			return fmt.Errorf("missing required %v variable", k)
+		}
+	}
+	if len(c.DeploymentFiles) == 0 {
+		return fmt.Errorf("missing deployment file(s)")
+	}
+	return nil
+}
+
+// ClusterCreate create a new cluster or applies changes to an existing cluster.
+func (c *KIND) ClusterCreate(*kingpin.ParseContext) error {
+	for _, deployment := range c.kindResources {
+		CreateWithConfigFile := cluster.CreateWithRawConfig(deployment.Content)
+
+		err := c.kindProvider.Create(c.DeploymentVars["CLUSTER_NAME"], CreateWithConfigFile)
+		if err != nil {
+			log.Fatalf("creating cluster err:%v", err)
+		}
+	}
+	return nil
+}
+
+// ClusterDelete deletes a k8s cluster.
+func (c *KIND) ClusterDelete(*kingpin.ParseContext) error {
+	err := c.kindProvider.Delete(c.DeploymentVars["CLUSTER_NAME"], homedir.HomeDir()+"/.kube/config")
```

Comment:
I think it'll be good to have a kubeconfig field in the KIND struct to store the location to the kubeconfig file and use it wherever the path to kubeconfig is required. Should the HomeDir method work when there is no kubeconfig present.(?) if we return it when creating the `New` function.